.github/workflows/src/summarize-checks.js

```diff
@@ -0,0 +1,827 @@
+ * @param {CheckMetadata[]} failingReqChecksInfo - Failing required checks info
+ * @param {CheckMetadata[]} failingFyiChecksInfo - Failing FYI checks info
+ * @param {RequiredLabelRule[]} violatedRequiredLabelsRules - Violated required label rules
+ * @returns {string} The body content HTML
+ */
+function getCommentBody(
+  requirementsMet,
+  anyBlockerPresent,
+  anyFyiPresent,
+  failingReqChecksInfo,
+  failingFyiChecksInfo,
+  violatedRequiredLabelsRules,
+) {
+  let bodyProper = "";
+
+  if (anyBlockerPresent || anyFyiPresent) {
+
+    if (anyBlockerPresent) {
+      bodyProper += getBlockerPresentBody(failingReqChecksInfo, violatedRequiredLabelsRules);
+    }
+
+    if (anyBlockerPresent && anyFyiPresent) {
+      bodyProper += "<br/>";
+    }
+
+    if (anyFyiPresent) {
+      bodyProper += getFyiPresentBody(failingFyiChecksInfo);
+      if (!anyBlockerPresent) {
+        bodyProper += `If you still want to proceed merging this PR without addressing the above failures, ${diagramTsg(4, false)}.`;
+      }
+    }
+  } else if (requirementsMet) {
+    bodyProper =
+      `✅ All automated merging requirements have been met! ` +
+      `To get your PR merged, see <a href="https://aka.ms/azsdk/specreview/merge">aka.ms/azsdk/specreview/merge</a>.`;
+  } else {
+    bodyProper =
+      "⌛ Please wait. Next steps to merge this PR are being evaluated by automation. ⌛";
+  }
+  return bodyProper;
+}
+
+/**
+ * Gets the body content when blockers are present
+ * @param {CheckMetadata[]} failingRequiredChecks - Failing required checks
+ * @param {RequiredLabelRule[]} violatedRequiredLabelsRules - Violated required label rules
+ * @returns {string} The blocker present body HTML
+ */
+function getBlockerPresentBody(failingRequiredChecks, violatedRequiredLabelsRules) {
+  const failingRequiredChecksNextStepsText = buildFailingChecksNextStepsText(
+    failingRequiredChecks,
+    "required",
+  );
+  const violatedReqLabelsRulesNextStepsText = buildViolatedLabelRulesNextStepsText(
+    violatedRequiredLabelsRules,
+  );
+  return (
+    "Next steps that must be taken to merge this PR: <br/>" +
+    "<ul>" +
+    violatedReqLabelsRulesNextStepsText +
+    failingRequiredChecksNextStepsText +
+    "</ul>"
+  );
+}
+
+/**
+ * Gets the body content when FYI issues are present
+ * @param {CheckMetadata[]} failingFyiChecksInfo - Failing FYI checks info
+ * @returns {string} The FYI present body HTML
+ */
+function getFyiPresentBody(failingFyiChecksInfo) {
+  return (
+    "Important checks have failed. As of today they are not blocking this PR, but in near future they may.<br/>" +
+    "Addressing the following failures is highly recommended:<br/>" +
+    "<ul>" +
+    buildFailingChecksNextStepsText(failingFyiChecksInfo, "FYI") +
+    "</ul>"
+  );
+}
+
+/**
+ * Builds next steps text for failing checks
+ * @param {CheckMetadata[]} failingChecks - Array of failing checks
+ * @param {"required" | "FYI"} checkKind - Kind of check (required or FYI)
```

Comment:
Could make this an "enum" if you care enough:

https://github.com/Azure/azure-rest-api-specs/blob/e877ba67daddbd28a0f868cded185a6ee395fa76/.github/workflows/src/label.js#L3-L11

Reply:
Honestly. I think the checkkind is a bit overkill anyway. It'll likely go away.